src/VideoCommand.cc

```diff
@@ -0,0 +1,352 @@
+/**
+ * @file   VideoCommand.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017 Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+#include <iostream>
+#include <fstream>
+
+#include "VideoCommand.h"
+#include "VDMSConfig.h"
+#include "defines.h"
+
+using namespace VDMS;
+
+VideoCommand::VideoCommand(const std::string &cmd_name):
+    RSCommand(cmd_name)
+{
+}
+
+void VideoCommand::enqueue_operations(VCL::Video& video, const Json::Value& ops)
+{
+    // Correct operation type and parameters are guaranteed at this point
+    for (auto& op : ops) {
+        const std::string& type = get_value<std::string>(op, "type");
+         std::string unit ;
+        if (type == "threshold") {
+            video.threshold(get_value<int>(op, "value"));
+
+        }
+        else if (type == "interval") {
+
+            video.interval(
+                VCL::Video::FRAMES,
+                get_value<int>(op, "start"),
+                get_value<int>(op, "stop"),
+                get_value<int>(op, "step"));
+
+        }
+        else if (type == "resize") {
+             video.resize(get_value<int>(op, "height"),
+                           get_value<int>(op, "width") );
+
+        }
+        else if (type == "crop") {
+            video.crop(VCL::Rectangle (
+                        get_value<int>(op, "x"),
+                        get_value<int>(op, "y"),
+                        get_value<int>(op, "width"),
```

Comment:
Inconsistent with the wiki page. Also, the wiki page has some JSON formatting issues in example. For a crop, number frames is missing and it is called w and h which is cool. I am not clear on the x,y,z naming.

Reply:
same as above.